src/Paginator/Adapter/AbstractAdapter.php

```diff
@@ -17,58 +17,75 @@
 use Phalcon\Paginator\Repository;
 use Phalcon\Paginator\RepositoryInterface;
 
+/**
+ * Phalcon\Paginator\Adapter\AbstractAdapter
+ */
 abstract class AbstractAdapter implements AdapterInterface
 {
+    /**
+     * Configuration of paginator
+     *
+     * @var array
+     */
+    protected array $config;
+
     /**
      * Number of rows to show in the paginator. By default is null
      *
-     * @var int
+     * @var int|null
      */
     protected ?int $limitRows = null;
 
     /**
      * Current page in paginate
      *
-     * @var int
+     * @var int|null
      */
     protected ?int $page = null;
 
     /**
      * Repository for pagination
      *
-     * @var RepositoryInterface
+     * @var RepositoryInterface|null
      */
-    protected RepositoryInterface $repository;
+    protected ?RepositoryInterface $repository = null;
 
     /**
      * Phalcon\Paginator\Adapter\AbstractAdapter constructor
      *
      * @param array $config
+     *
+     * @throws Exception
      */
-    public function __construct(
-        protected array $config
-    ) {
-        $this->repository = new Repository();
```

Comment:
This needs to be up there - the protected part makes use of the property and the `new Repository()` saves the `if` statement further down when you get the repository.